### PR TITLE
refactor: replace `uncordonNode` with controllers

### LINKED ISSUE #7233: Refactor `uncordonNode` & `labelControlPlane` tasks
### `labelControlPlane`

Only on controlplane machines.

1. Labels the `Node` as controlplane.
2. Taint the `Node` as unschedulable (if not disabled in the config).

### `uncordonNode`

1. Wait for the `Node` to be `Ready`.
2. Remove the Talos taint set on upgrade/etc. (when cordoning the node).

### Controlplane Label

Shouldn't ever block the boot sequence.

Can be implemented directly by injecting a proper label to the `NodeLabelsApplyController`.

### Tainting Controlplane Machines

Shouldn't ever block the boot sequence.

```mermaid
flowchart LR

  NodeName([NodeName ready]) --> WatchNode[/watch Node resource/]
  WatchNode --> Node_is_Ready([Node is Ready])

```

### PR DESCRIPTION
**Pull request by smira**
Fixes #7233

Waiting for node readiness now happens in the `MachineStatus` controller which won't mark the node as ready until Kubernetes `Node` is ready.

Handling cordoning/uncordining happens wiht help of additional resource in `NodeApplyController`.

New controller provides reactive `NodeStatus` resource to see current status of Kubernetes `Node`.

### CODE REVIEW

File: internal/app/machined/pkg/controllers/k8s/node_apply_test.go [hunk 1]
```diff
@@ -109,7 +112,11 @@ func TestApplyLabels(t *testing.T) {
 			},
 		},
 	} {
+		tt := tt
```
Comment (frezbo) [inline internal/app/machined/pkg/controllers/k8s/node_apply_test.go:115]: we don't need this with go1.20 right??

File: internal/app/machined/pkg/controllers/k8s/node_cordoned_spec.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+			return fmt.Errorf("error getting config: %w", err)
+		}
+
+		var shouldCordon bool
+
+		switch status.TypedSpec().Stage { //nolint:exhaustive
+		case runtime.MachineStageShuttingDown, runtime.MachineStageUpgrading, runtime.MachineStageResetting:
+			shouldCordon = true
+		case runtime.MachineStageBooting, runtime.MachineStageRunning:
```
Comment (frezbo) [inline internal/app/machined/pkg/controllers/k8s/node_cordoned_spec.go:76]: `MachineStageBooting` is okay to uncordon?

File: internal/app/machined/pkg/controllers/k8s/node_status.go [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+		nodename, err := safe.ReaderGetByID[*k8s.Nodename](ctx, r, k8s.NodenameID)
+		if err != nil {
+			if !state.IsNotFoundError(err) {
+				return fmt.Errorf("error getting nodename: %w", err)
+			}
+
+			continue
+		}
+
+		if nodename.TypedSpec().SkipNodeRegistration {
```
Comment (frezbo) [inline internal/app/machined/pkg/controllers/k8s/node_status.go:109]: :cool:

File: internal/app/machined/pkg/controllers/k8s/node_cordoned_spec.go (hunk 2 above)
Comment (smira) [reply internal/app/machined/pkg/controllers/k8s/node_cordoned_spec.go:76]: yes, we cordon before reset/shutdown to effectively drain the load. !ready nodes are not schedulable

File: internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer.go [hunk 4]
```diff
@@ -276,10 +276,6 @@ func (*Sequencer) Boot(r runtime.Runtime) []runtime.Phase {
 	).Append(
 		"startEverything",
 		StartAllServices,
-	).AppendWhen(
-		r.State().Platform().Mode() != runtime.ModeContainer && !r.Config().Machine().Kubelet().SkipNodeRegistration(),
```
Comment (frezbo) [inline internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer.go:280]: I didn't see `runtime.ModeContainer` handled elsewhere :thinking:

Comment (smira) [reply internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer.go:280]: I wasn't sure why it matters in container mode.

**Review by frezbo — APPROVED** [review]

File: internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer.go (hunk 4 above)
Comment (frezbo) [reply internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer.go:280]: not sure either, maybe kubelet handles it differently?

Comment (smira) [reply internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer.go:280]: nope, I don't think it cares

Comment (smira) [reply internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer.go:280]: looks like tests are flaky

**Pull request merged**